### PR TITLE
tools/all_syscalls & meson depends simplification, program-tests option

### PR DESCRIPTION
**Pull request by apprehensions**
using sh removes the bash dependencies, for those who don't have
bash on their system, and replaced the GNU awk with a portable
grep and sed.

as for the meson, I've done the same and used the beginner meson book. since all_syscalls utility will always right syscalls.h, theres no point to have it be an output and launched by bash, if meson has this feature already called `run_command`.

### CODE REVIEW

**Comment by karelzak** [comment]
@t-8ch any objection & comment?

File: libmount/meson.build [hunk 1]
```diff
@@ -131,20 +131,22 @@ libmount_test_src_override = {
...
+if program_tests
+  foreach libmount_test: libmount_tests
+    test_name = 'test_mount_' + libmount_test
+    exe = executable(
+      test_name,
+      'src/' + libmount_test_src_override.get(libmount_test, libmount_test) + '.c',
+      include_directories : [dir_include, dir_libblkid],
+      link_with : [lib__mount, lib_common, lib_blkid_static],
+      dependencies : lib__mount_deps,
+      c_args : ['-DTEST_PROGRAM'],
```
Comment (t-8ch) [inline libmount/meson.build:143]: Why not use the same `build_by_default` logic that is used elsewhere in the patch?

File: tools/all_syscalls [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
```
Comment (t-8ch) [inline tools/all_syscalls:1]: This breaks the build during CI.

**Review by t-8ch — CHANGES_REQUESTED** [review]

**Review by t-8ch — APPROVED** [review]

File: tools/all_syscalls (hunk 2 above)
Comment (karelzak) [reply tools/all_syscalls:1]: We have almost everywhere bash ...

Comment (apprehensions) [reply tools/all_syscalls:1]: I'm aware, i only left the important utilities (small ones, ones that dont require bash all that much), but focused on the one that is only required at build-time, the rest aren't required at build-time.

Comment (t-8ch) [reply tools/all_syscalls:1]: Unfortunately `set -o pipefail` is not POSIX and it breaks again on systems where `/bin/sh` is not bash.

Comment (apprehensions) [reply tools/all_syscalls:1]: > Unfortunately set -o pipefail is not POSIX
```
# pipefail is part of modern POSIX
```

Comment (t-8ch) [reply tools/all_syscalls:1]: It still breaks our build. And users *will* encounter this issue.

Comment (masatake) [reply tools/all_syscalls:1]: I'm encountering this issue at https://github.com/util-linux/util-linux/pull/2394.

image

**Comment by masatake** [comment]
I cannot understand the decision to merge this pull request.

The tests failed on many platforms. Isn't the result of testing a part of the criteria for merging?

File: tools/all_syscalls (hunk 2 above)
Comment (apprehensions) [reply tools/all_syscalls:1]: unfortunately there is no realistic or clean way to have a pipefail-like feature in a script like this, you either have to:
1. disable the building/requirement of the all_syscalls via an available option
2. update your shell to latest POSIX

**Comment by karelzak** [comment]
Sorry, reverted the patch.

**Pull request merged**